File: 92-reverse-linked-list-ii/92-reverse-linked-list-ii.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseBetween(self, head: Optional[ListNode], left: int, right: int) -> Optional[ListNode]:
        temp = ListNode()
        temp.next = head
        
        tail = temp
        index = 0
        
        while index < left - 1:
            tail = tail.next
            index += 1
        
        node = tail.next
        stack = []
        
        while index < right:
            stack.append(node)
            node = node.next
            index += 1
        
        while stack:
            tail.next = stack.pop()
            tail = tail.next
        
        tail.next = node
        
        return temp.next

File: 92-reverse-linked-list-ii/92-reverse-linked-list-ii.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func reverseBetween(head *ListNode, left int, right int) *ListNode {
    temp := &ListNode{Next: head}
    
    tail := temp
    index := 0
    
    for index < left - 1 {
        tail = tail.Next
        index++
    }
    
    node := tail.Next
    stack := make([]*ListNode, right - left + 1)
    
    for i := 0; i < len(stack); i++ {
        stack[i] = node
        node = node.Next
    }
    for i := len(stack) - 1; i >= 0; i-- {
        tail.Next = stack[i]
        tail = tail.Next
    }
    
    tail.Next = node
    
    return temp.Next
}